lib/rules/declaration-block-no-redundant-longhand-properties/index.mjs

```diff
@@ -272,12 +278,52 @@ const rule = (primary, secondaryOptions) => {
 				const prop = decl.prop.toLowerCase();
 				const unprefixedProp = vendor.unprefixed(prop);
 				const prefix = vendor.prefix(prop);
+
+				// Check if this is a shorthand property and track it
+				const shorthandLonghands = longhandSubPropertiesOfShorthandProperties.get(
+					/** @type {import('stylelint').ShorthandProperties} */ (unprefixedProp),
+				);
+
+				if (shorthandLonghands) {
+					// This is a shorthand property, store it with its value
+					declaredShorthands.set(prop, decl.value.trim());
+				}
+
 				const shorthandProperties = longhandToShorthands.get(unprefixedProp);
 
 				if (!shorthandProperties) {
 					return;
 				}
 
+				// Check if a covering shorthand was declared earlier with the same value
+				for (const shorthandProperty of shorthandProperties) {
```

Comment:
[suggestion] I'd like to avoid increasing a `for` loop. Can you integrate the new loop into the existing one?

https://github.com/stylelint/stylelint/blob/01786e9e5281c153ff6a983d770b749fa0f8252f/lib/rules/declaration-block-no-redundant-longhand-properties/index.mjs#L327